Fix gem build warnings

deivid-rodriguez opened the pull request:
#### :tophat: What? Why?

This PR reviews gemspecs to remove `gem build` warnings. Since I was at it, I also bumped gems in `Gemfile.lock`.

* Changes our dependency version specifiers to use strict semantic versions. The exception is when we know the dependency doesn't follow semver in which case I set an upper bound for the dependency that should be bumped manually, for example, `rails` or `foundation-rails`. In addition to fixing a handful of `gem build` warnings, this should make end apps less likely to run into version conflicts since the dependencies are a bit more relaxed.

* I normalized gem summary and descriptions to not be duplicated.

* I added a missing homepage in the main gemspec.

#### :pushpin: Related Issues
_None_.

#### :clipboard: Subtasks
_None_.

### :camera: Screenshots (optional)
_None_.

mrcasals commented:
@deivid-rodriguez is this warning new?

```
WARNING:  pessimistic dependency on active_link_to (~> 1.0.0) may be overly strict
  if active_link_to is semantically versioned, use:
    add_runtime_dependency 'active_link_to', '~> 1.0', '>= 1.0.0'
```

I haven't seen it before 😕

deivid-rodriguez commented:
@mrcasals No, I think it's always been there when running `gem build decidim-core.gemspec` (or others). I think we haven't seen it because we only run `gem build` under the hood for the `rake release` task and that task seems to swallow those warnings...

mrcasals commented:
@deivid-rodriguez can you fix the conflicts on the `Gemfile.lock` file, please?

deivid-rodriguez commented:
Rebased & fixed conflicts, and added a few extra changes to make `decidim-debates` gemspec consistent with the rest of the gemspec files.

josepjaume commented on decidim-budgets/decidim-budgets.gemspec:
```diff
@@ -15,15 +15,15 @@ Gem::Specification.new do |s|
   s.name = "decidim-budgets"
-  s.summary = "A budgets component for decidim's participatory processes."
-  s.description = s.summary
+  s.summary = "Decidim budgets plugin"
```
We should be using the `module` nomenclature instead of `plugin`

josepjaume commented:
Super nice!

deivid-rodriguez commented:
I removed the "plugin" word from gemspecs and a couple other places.

I also sneaked in some extra changes (to bound `rubocop-rspec` to match codeclimate's version and also add a couple other tiny changes related to it). I was planning to open a separate PR but since I'm pushing to this branch again and they are somehow related I added them.
